pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -303,6 +316,23 @@ def newpkgcb(pkgId, name, arch):
             result = await downloader.run()
             metadata_pb.increment()
 
+            repomd_path = result.path
+            repomd = cr.Repomd(repomd_path)
+
+            if optimize_sync:
+                if self.repository.latest_version().number > 1:
+                    optimize_data = dict(message='Optimizing Sync', code='optimizing.zync')
+                    with ProgressReport(**optimize_data) as optimize_pb:
+                        if repomd.revision <= self.repository.revision_number:
```

Comment:
All of these conditions should not be nested. You need to perform and/or operations on them in one long condition.